File: app/src/main/java/fr/nihilus/xenobladechronicles/datastore/Converters.java
package fr.nihilus.xenobladechronicles.datastore;

import android.arch.persistence.room.TypeConverter;

import fr.nihilus.xenobladechronicles.model.Area;

class Converters {

    @TypeConverter
    public static Area fromName(String value) {
        return value == null ? null : Area.valueOf(Area.class, value);
    }

    @TypeConverter
    public static String areaToName(Area area) {
        return area == null ? null : area.name();
    }
}

File: app/src/main/java/fr/nihilus/xenobladechronicles/datastore/Converters.kt
package fr.nihilus.xenobladechronicles.datastore

import androidx.room.TypeConverter
import fr.nihilus.xenobladechronicles.model.Area

internal class Converters {

    @TypeConverter
    fun fromName(areaCodeName: String?): Area? = areaCodeName?.let(Area::valueOf)

    @TypeConverter
    fun areaToName(area: Area?): String? = area?.name
}
